prov/sm2/src/sm2_progress.c

```diff
@@ -293,10 +438,93 @@ static int sm2_progress_recv_msg(struct sm2_ep *ep,
 {
 	struct fid_peer_srx *peer_srx = sm2_get_peer_srx(ep);
 	struct fi_peer_rx_entry *rx_entry;
+	struct sm2_cma_data *cma_data;
 	struct sm2_av *sm2_av;
 	fi_addr_t addr;
+	uint64_t comp_flags;
 	int ret;
 
+	/* TODO - Switch on protocol before switching on op to avoid messy
+	 * checks like this */
+
+	/* We don't want to look for rx entries for xfer entries with
+	 * SM2_CMA_HOST_TO_DEV and SM2_CMA_HOST_TO_DEV_ACK flags*/
+	if (xfer_entry->hdr.proto == sm2_proto_cma) {
+		if (xfer_entry->hdr.proto_flags & SM2_CMA_HOST_TO_DEV) {
```

Comment:
I would like it if lines 453 -> 522 lived in a static inline function like sm2_host_to_dev_progress (feel free to come up with better name).  This is already messy... no need to make the common path unreadable.

Reply:
Done